### PR TITLE
chore: prohibit barrels

### PR DESCRIPTION
**Pull request by zzmp**
## Description

Lints against the typescript barrel pattern (`export *`) because it prevents checks for unused code and it reduces tree-shaking (because the barrel is considered a side effect and always bundled).

This eliminates some unused nft code from first load, including the entirety of the @opensea dependency, altogether reducing the initial bundle by about 12.5%.

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export * from './addresses'
-export * from './escapeRegExp'
-export * from './getContract'
+export { isAddress, shortenAddress } from './addresses'
```
Comment (cbachmeier) [inline src/utils/index.ts:1]: this feels like a good barrel compromise

**Review by cbachmeier — APPROVED** [review]
Good night sweet prince 🫡

File: .eslintrc.js [hunk 2]
```diff
@@ -70,6 +70,13 @@ module.exports = {
             ],
           },
         ],
+        'no-restricted-syntax': [
+          'error',
+          {
+            selector: ':matches(ExportAllDeclaration)',
+            message: 'Barrel exports bloat the bundle size.',
```
Comment (cbachmeier) [inline .eslintrc.js:77]: nit: could also add a note about tree-shaking

**Review by JackShort — APPROVED** [review]
actually viewed every file

**Pull request merged**